File: src/main/java/ordilov/lissn/security/userinfo/GoogleOAuth2UserInfo.java
package ordilov.lissn.security.userinfo;

import lombok.Getter;
import lombok.ToString;
import ordilov.lissn.member.domain.AuthProvider;
import org.springframework.security.oauth2.core.user.OAuth2User;

@Getter
@ToString
public class GoogleOAuth2UserInfo extends OAuth2UserInfo {

  public GoogleOAuth2UserInfo(OAuth2User oAuth2User) {
    this.provider = AuthProvider.google;
    this.id = oAuth2User.getAttributes().get("sub").toString();
    this.name = oAuth2User.getAttributes().get("name").toString();
    this.email = oAuth2User.getAttributes().get("email").toString();
    this.picture = oAuth2User.getAttributes().get("picture").toString();
  }
}

File: src/main/java/ordilov/lissn/security/userinfo/GoogleOAuth2UserInfo.kt
package ordilov.lissn.security.userinfo;

import ordilov.lissn.member.domain.AuthProvider
import org.springframework.security.oauth2.core.user.OAuth2User

class GoogleOAuth2UserInfo(
    id: String,
    name: String,
    email: String,
    picture: String,
    provider: AuthProvider
) : OAuth2UserInfo(
    id, name, email, picture, provider
) {

    constructor(oAuth2User: OAuth2User) : this(
        oAuth2User.attributes["sub"] as String,
        oAuth2User.attributes["name"] as String,
        oAuth2User.attributes["email"] as String,
        oAuth2User.attributes["picture"] as String,
        AuthProvider.google
    )
}